### ISSUE TITLE
exclude option not working.

### ISSUE THREAD

**Issue by amitdhiman000** [webpack-contrib/eslint-loader#322]
```
{
      enforce: 'pre',
      test: /\.js$/,
      exclude: /(node_modules)/,
      include: path.resolve('src'),
      loader: 'eslint-loader',
}
```

Problem:
1.) none of the options (include or exclude) option is working.
2.) It is not honouring **.eslintignore** file,folders.
seeing errors like:
ERROR in ./src/index.js
Module build failed (from ./node_modules/eslint-loader/dist/cjs.js):
react_app/node_modules/fs-extra/lib/mkdirs/make-dir.js:86
      } catch {
              ^

linter is going inside **node_modules**, showing errors for node_modules code.

Expected:
exclude option should exclude the paths specified.
include option is expected to include the files, folders specified.

P.S.
This problem is specially happening with eslint-loader : 4.0.0
temporary solution is to use  eslint-loader: 3.0.4

**Comment by ricardogobbosouza** [comment]
Hi @amitdhiman000 
Please provide a repository with the issue.

**Comment by Ghostfly** [comment]
Try to upgrade :
- https://github.com/Ghostfly/ghostfly.dev (eslint-loader) => Boom
- https://github.com/Ghostfly/persin-conseil => Boom
I think that 4.0.0 is buggy.

**Comment by ricardogobbosouza** [comment]
Hi @Ghostfly 
Resolved on [v4.01](https://github.com/webpack-contrib/eslint-loader/releases/tag/v4.0.1)

**Comment by Ghostfly** [comment]
Still there : 
<img width="663" alt="Capture d’écran 2020-04-24 à 20 57 23" src="https://user-images.githubusercontent.com/5880133/80247373-3a3fb680-866e-11ea-90f6-12a665e79e48.png">

**Comment by ricardogobbosouza** [comment]
@Ghostfly its work for me

**Comment by ricardogobbosouza** [comment]
try re-installing
`rm -rf node_modules package-lock.json && npm install`

**Comment by Ghostfly** [comment]
Already tried (and now : 
<img width="663" alt="Capture d’écran 2020-04-24 à 20 57 23" src="https://user-images.githubusercontent.com/5880133/80250155-8d683800-8673-11ea-8795-54df63988272.png"> again.)

Are you running on macOS ?

**Comment by ricardogobbosouza** [comment]
Bug on `fs-extra`

https://github.com/jprichardson/node-fs-extra/issues/771

**Comment by ricardogobbosouza** [comment]
I will revert to version 8.1.0 of fs-extra

**Comment by ricardogobbosouza** [comment]
Hi @Ghostfly
Try v4.0.2

**Comment by Ghostfly** [comment]
Working perfectly ! 👍 🥇

**Comment by amitdhiman000** [comment]
@ricardogobbosouza 
by looking at discussion, it looks like it resolved with v4.0.2.
I was not possible to share the repo, it is on my organization's internal git

**Comment by amitdhiman000** [comment]
> Hi @amitdhiman000
> Please provide a repository with the issue.

@ricardogobbosouza
by looking at discussion, it looks like it resolved with v4.0.2.
I was not possible to share the repo, it is on my organization's internal git
